### PR TITLE
Clean panner API

### PR DESCRIPTION
**Pull request by b-ma**
Finally took the time to check the error I have seen trying to generate the bindings. So here we are:
- A bit here, with the `impl Default` for some options
- A bit on the bindings side to ignore the deprecated functions that are not implemented here

### CODE REVIEW

File: src/node/panner.rs [hunk 1]
```diff
@@ -103,10 +116,45 @@ impl Default for PannerOptions {
...
+            channel_config: ChannelConfigOptions {
+                count: 2,
+                count_mode: ChannelCountMode::ClampedMax,
+                interpretation: ChannelInterpretation::Speakers,
+            },
         }
     }
 }
 
+/// Assert that the channel count is valid for the StereoPannerNode
```
Comment (orottier) [inline src/node/panner.rs:128]:
```suggestion
/// Assert that the channel count is valid for the PannerNode
```

File: src/node/panner.rs [hunk 2]
```diff
@@ -103,10 +116,45 @@ impl Default for PannerOptions {
...
+///
+#[track_caller]
+#[inline(always)]
+fn assert_valid_channel_count(count: usize) {
+    if count > 2 {
+        panic!("NotSupportedError: PannerNode channel count cannot be greater than two");
+    }
+}
+
+/// Assert that the channel count is valid for the StereoPannerNode
```
Comment (orottier) [inline src/node/panner.rs:143]:
```suggestion
/// Assert that the channel count is valid for the PannerNode
```

File: src/node/panner.rs [hunk 3]
```diff
@@ -262,23 +310,35 @@ impl AudioNode for PannerNode {
         1
     }
 
-    fn set_channel_count(&self, v: usize) {
-        if v > 2 {
-            panic!("NotSupportedError: PannerNode channel count cannot be greater than two");
-        }
-        self.channel_config.set_count(v);
+    // same limitations as for the PannerNode
```
Comment (orottier) [inline src/node/panner.rs:313]:
```suggestion
    // same limitations as for the StereoPannerNode
```

**Review by orottier — COMMENTED** [review]
Great fixes

**Comment by b-ma** [comment]
Thanks for fixing the issues in the comments!

**Pull request merged**